pkg/nfd-master/nfd-master.go

```diff
@@ -346,6 +395,51 @@ func removeLabels(n *api.Node, labelNames []string) {
 	}
 }
 
+// getExtendedResourceOps returns a slice of operations to perform on the node status
+func getExtendedResourceOps(n *api.Node, extendedResources ExtendedResources) []statusOp {
+	var statusOps []statusOp
+	addStatusOp := func(verb string, resource string, path string, value string) {
+		res := strings.ReplaceAll(resource, "/", "~1")
+		op := statusOp{verb, "/status/" + path + "/" + res, value}
+		statusOps = append(statusOps, op)
+	}
+
+	oldResources := strings.Split(n.Annotations[AnnotationNs+"extended-resources"], ",")
+
+	// figure out which resources to remove
+	for _, resource := range oldResources {
+		// provided that the resource exists
+		if _, ok := n.Status.Capacity[api.ResourceName(resource)]; ok {
+			// check if the related label is gone (empty value), or the ext resource requirement is gone
+			value, extResExists := extendedResources[resource]
+			if !extResExists || value == "" {
+				addStatusOp("remove", resource, "capacity", "")
+				addStatusOp("remove", resource, "allocatable", "")
+			}
+		}
+	}
+
+	// figure out which resources to replace and which to add
+	for resource, value := range extendedResources {
+		// check that there is such a label (value is non-empty)
+		if value != "" {
```

Comment:
With the change I proposed in `filterFeatureLabels` you could remove this check, I think